src/snyk/cli/services/cliService.ts

```diff
@@ -66,4 +86,26 @@ export abstract class CliService<CliResult> extends AnalysisStatusProvider {
 
     return args;
   }
+
+  public async isChecksumCorrect(cliPath: string): Promise<boolean> {
+    if (!_.isUndefined(this.verifiedChecksumCorrect)) {
```

Comment:
We verify checksum only at plugin start (oss init) and then reuse result in next test() calls. Shouldn't we do those check every test() invokation?

Reply:
Is there any additional benefit if we do it on every test() invocation?

Reply:
we call isChecksumCorrect() every test()
Then we either should not do that (as we keep checking against saved result) or make calculation for each invocation.
I would prefer the last one as binary could be corrupted/replaced since plugin start.

Reply:
@ArtsiomCh I assume that the binary won't be modified during the runtime. It's quite a rare case before the startup to be modified (we cannot always rely on it), and even rarer during runtime. As checksum calculation takes time, I cache the value for now. I propose to leave it out until we see it's worth doing on every test run.